src/odemis/model/_futures.py

```diff
@@ -489,7 +489,8 @@ def set_running_or_notify_cancel(self):
             # Set start to current time and end to estimated endtime (difference between current end and start)
             start, end = self.get_progress()
             startt = time.time()
-            endt = startt + end - start
+            endt = startt + (end - start)
```

Comment:
Can you add this as a comment here, so that nobody ends up removing the brackets at some point?

`Force the order of computation to reduce floating point error.`